packages/buidler-core/src/internal/buidler-evm/jsonrpc/handler.ts

```diff
@@ -106,7 +106,9 @@ export default class JsonRpcHandler {
           rpcReq.method === "eth_subscribe" &&
           isSuccessfulJsonResponse(rpcResp)
         ) {
-          subscriptions.push(rpcResp.result.id);
+          try {
+            subscriptions.push((rpcResp.result as any).id);
+          } catch {}
```

Comment:
Is this accidental?

Reply:
This is a result of proper types being applied to the jsonRpc responses